ui/apps/platform/src/Containers/Collections/errorUtils.ts

```diff
@@ -0,0 +1,76 @@
+import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
+
+type CollectionSaveErrorType =
+    | 'CollectionLoop'
+    | 'DuplicateName'
+    | 'InvalidRule'
+    | 'EmptyCollection'
+    | 'UnknownError';
+
+export type CollectionSaveError =
+    | {
+          type: 'CollectionLoop';
+          message: string;
+          details?: string;
+          loopId: string | undefined;
+      }
+    | {
+          type: Exclude<CollectionSaveErrorType, 'CollectionLoop'>;
+          message: string;
+          details?: string;
+      };
+
+/**
+ * Given an error, attempt to categorize and provide more information to the user. These are
+ * errors that cannot be detected by general `yup` validation and often can only be detected
+ * by the server.
+ *
+ * @param err An instance of an Error
+ * @return A categorized error specific to collections
+ */
+export function parseSaveError(err: Error): CollectionSaveError {
+    const rawMessage = getAxiosErrorMessage(err);
+
+    if (/create a loop/.test(rawMessage)) {
+        const errorRegex = /^edge between '[0-9a-fA-F-]*' and '(?<loopId>[0-9a-fA-F-]*)'/;
+        const matches = errorRegex.exec(rawMessage);
+        const loopId = matches?.groups?.loopId;
+        return {
+            type: 'CollectionLoop',
+            message: 'An attached collection has created a loop, which is not supported',
+            details: 'Detach the invalid collection and try saving again',
+            loopId,
+        };
+    }
+
+    if (
+        // Error for collection save
+        /collections must have non-empty, unique `name` values/.test(rawMessage) ||
+        // Error for collection update
+        /name already in use/.test(rawMessage)
+    ) {
+        return { type: 'DuplicateName', message: 'Collection name values must be unique' };
```

Comment:
The `message` could be more parallel to **name is a required field**
* Although the message I quoted is yup default? Because **name** lowercase from key instead of sentence case from field label?
* How about drop **values** to get either **Collection names must be unique** if that helps understanding or **Name must be unique** for most direct relationship with label